core/src/main/java/pl/allegro/tech/embeddedelasticsearch/IndexRequest.java

```diff
@@ -0,0 +1,46 @@
+package pl.allegro.tech.embeddedelasticsearch;
+
+public class IndexRequest {
+
+    private final String indexName;
+    private final String indexType;
+    private final String id;
+    private final String routing;
+    private final String json;
+
+    public IndexRequest(String indexName, String indexType, String json) {
+        this(indexName, indexType, json, null, null);
+    }
+
+    public IndexRequest(String indexName, String indexType, String json, String id) {
+        this(indexName, indexType, json, id, null);
+    }
+
+    public IndexRequest(String indexName, String indexType, String json, String id, String routing) {
```

Comment:
Ok, we have IndexRequest, but it still takes up to 5 parameters, which still is too much :) Please add builder for this class and leave only one private constructor